src/user/userReducer.js

```diff
@@ -12,13 +12,22 @@ const initialState = {
   fileUploadError: null,
   filesFetchError: null,
   filesFetchIsLoading: true,
+  recommendations: [],
   following: {
     list: [],
     pendingFollows: [],
     isFetching: false,
   },
 };
 
+// filterRecommendations generates a random list of `count` recommendations
+// include users followed by the current user.
+const filterRecommendations = (following, count = 5) => {
+  const usernames = Object.values(following);
+  const filtered = usernames.length > 0 ? people.filter(p => !usernames.includes(p)) : people;
+  return filtered.sort(() => 0.5 - Math.random()).slice(0, count).map(r => ({ name: r }));
```

Comment:
I don't think that checking `usernames.length` is necessary. I think we can chain this to one expression like this:
`const filterRecommendations = (following, count = 5) => Object.values(following).filter(...).sort(...).slice(...).map(...)`
Variable `r` in map has not clear meaning so I think we can change it to `name` (so we can use object property literal shorthand like this: `map(name => ({ name }))`

Reply:
Good suggestions, especially the object property literal shorthand.

The reason for the `usernames.length` check is to not iterate over the `people` list (which is fairly large) an extra time if it's not necessary (if the followers list hasn't loaded yet or the user has no followers). Since it's only a list of 101 people at this time, I suppose it's not a huge deal to iterate it more than necessary.

Your suggestion for rewriting the function in the arrow syntax shorthand of `param => returnval` does look nice, but what you wrote is actually the other way around from how it is.

It's not `Object.values(following)` that we're filtering on - it's `people`. We want to say, "Take the list of interesting people and filter out anyone who the user already follows".

So if I rewrite the function as suggested, we get:
```
const filterRecommendations = (following, count = 5) =>
  people.filter(p => !Object.values(following).includes(p))
        .sort(() => 0.5 - Math.random())
        .slice(0, count)
        .map(name => ({name}));
```

This I have a problem with, because in order to gain the syntactic sugar of `param => returnval`, we have to recompute Object.values(following) for every item of the `people` list (101 times), and I just don't think that's a good idea. In the case that a user follows 1,000 or more users, that's could get slow, and I definitely don't want reducer code to be slow.

So here's what I'm proposing:
- No more usernames.length check
- Sacrifice the syntactic sugar function definition in favor of only iterating the user's following list *once* instead of `people.length` times.
- Definitely clean up the `map` to use the literal shorthand

It will look like this:
```
const filterRecommendations = (following, count = 5) => {
  const usernames = Object.values(following);
  return people.filter(p => !usernames.includes(p))
        .sort(() => 0.5 - Math.random())
        .slice(0, count)
        .map(name => ({name}));
}
```

I've pushed the updated code. Let me know what you think.